src/main/java/com/bobocode/hashtable/HashTable.java

```diff
@@ -0,0 +1,142 @@
+package com.bobocode.hashtable;
+
+import java.util.Objects;
+
+/**
+ * A simple implementation of the Hash Table that allows storing a generic key-value pair. The table itself is based
+ * on the array of {@link Node} objects.
+ * <p>
+ * An initial array capacity is 16.
+ * <p>
+ * Every time a number of elements is equal to the array size that tables gets resized
+ * (it gets replaced with a new array that it twice bigger than before). E.g. resize operation will replace array
+ * of size 16 with a new array of size 32. PLEASE NOTE that all elements should be reinserted to the new table to make
+ * sure that they are still accessible  from the outside by the same key.
+ *
+ * @param <K> key type parameter
+ * @param <V> value type parameter
+ */
+public class HashTable<K, V> {
+
+    private static final int DEFAULT_CAPACITY = 16;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public HashTable() {
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
+
+    /**
+     * Puts a new element to the table by its key. If there is an existing element by such key then it gets replaced
+     * with a new one, and the old value is returned from the method. If there is no such key then it gets added and
+     * null value is returned.
+     *
+     * @param key   element key
+     * @param value element value
+     * @return old value or null
+     */
+    public V put(K key, V value) {
+        Objects.requireNonNull(key, "Null keys are not allowed!");
+        Objects.requireNonNull(value, "Null values are not allowed!");
+        checkInternalCapacity();
+        var bucket = getBucket(key, this.table.length);
+        V oldValue = null;
+        var old = findNode(key);
```

Comment:
@vromantsev you already have a bucked (an index), so there is no need to search by key